pkg/protocol/inactivity/inactivity.go

```diff
@@ -0,0 +1,88 @@
+package inactivity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-log/v2"
+
+	"github.com/keep-network/keep-core/pkg/net"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+	"github.com/keep-network/keep-core/pkg/protocol/state"
+)
+
+// SignedClaim represents information pertaining to the process of signing
+// an inactivity claim: the public key used during signing, the resulting
+// signature and the hash of the inactivity claim that was used during signing.
+type SignedClaim struct {
+	PublicKey []byte
+	Signature []byte
+	ClaimHash ClaimSignatureHash
+}
+
+type ClaimSigner interface {
+	SignClaim(claim *Claim) (*SignedClaim, error)
+	VerifySignature(signedClaim *SignedClaim) (bool, error)
+}
+
+type ClaimSubmitter interface {
+	SubmitClaim(
+		ctx context.Context,
+		memberIndex group.MemberIndex,
+		claim *Claim,
+		signatures map[group.MemberIndex][]byte,
+	) error
+}
+
+func Publish(
```

Comment:
Let's be more specific:
```suggestion
func PublishClaim(
```

Reply:
Done in 0e3b9e5a5e7bfe2b010ecf77509bdfe5536713a2.